File: fixtures/scale/java/large/src/cryptobench/ciphers/hill/HillSuite.java
package cryptobench.ciphers.hill;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class HillSuite implements CipherSuite {
    @Override
    public String name() {
        return "hill";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new HillCipher(HillKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : HillVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/dart/large/lib/ciphers/hill/hill_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'hill_cipher.dart';
import 'hill_key.dart';
import 'hill_vectors.dart';

class HillSuite implements CipherSuite {
  @override
  String name() => 'hill';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = HillCipher(HillKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in hillSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
